### ISSUE TITLE
DNS V2: recordset records are returned in a random order

### ISSUE THREAD

**Issue by kayrus** [terraform-providers/terraform-provider-openstack#627]
### Affected Resource(s)
- openstack_dns_recordset_v2

Designate test fails:

> --- FAIL: TestAccDNSV2RecordSet_ipv6 (46.57s)
>    testing.go:518: Step 0 error: Check failed: Check 3/4 error: openstack_dns_recordset_v2.recordset_1: Attribute 'records.0' expected "fd2b:db7f:6ae:dd8d::1", got "fd2b:db7f:6ae:dd8d::2"

**Comment by jtopjian** [comment]
That's a good catch. 

While OpenLab does not test Designate by default (you can trigger it by doing [`recheck designate`](https://www.terraform.io/docs/providers/openstack/index.html#recheck-designate), I test DNS in another cloud regularly and have not seen this problem.

I even checked my shell history to confirm I ran all tests before I submitted #568.

However, I just ran the `TestAccDNSV2RecordSet_ipv6` again and it failed. But then I ran it a second time and it passed:

```
2019/01/18 04:18:00 [DEBUG] OpenStack Response Body: {
  "action": "NONE",
  "created_at": "2019-01-18T04:17:16.000000",
  "description": "a record set",
  "id": "a59a7733-04ad-449c-b965-25bc3adcd54f",
  "name": "ACPTTEST-zone-s3fp1.com.",
  "records": [
    "fd2b:db7f:6ae:dd8d::1",
    "fd2b:db7f:6ae:dd8d::2"
  ],
  "status": "ACTIVE",
  "ttl": 3000,
  "type": "AAAA",
  "updated_at": null,
  "version": 1,
  "zone_id": "ad54b156-46ab-4252-9401-8de68167c101"
}
```

And then I ran it a third time and saw Designate reporting the correct order, but mid-way through the refresh func, Designate started sending the records back in a different order, meaning Designate decided to change the ordering.

I am hesitant about changing the `TypeList` to `TypeSet`. I imagine that a lot of users use the 0-based indexing to reference the records. Perhaps the `DiffSuppressFunc` will need further modified to suppress out of order diffs.

**Comment by jtopjian** [comment]
> DiffSuppressFunc works on elements of a list/set and not on the list/set as a whole, so further modification there isn't going to help.

This is incorrect. `DiffSuppressFunc` provides access to the full schema, but it's still not a valid solution.

I think I have a solution to resolve this, but will need to work on it. 

At the same time, I'm surprised that this issue hasn't been reported before. Specifying multiple IPs in a record set should be a common scenario.

**Comment by jtopjian** [comment]
This has been resolved in #636
